daemon/images/image.go

```diff
@@ -44,6 +162,19 @@ func (i *ImageService) GetImage(refOrID string, platform *specs.Platform) (retIm
 		// Note that `platforms.Only` will fuzzy match this for us
 		// For example: an armv6 image will run just fine an an armv7 CPU, without emulation or anything.
 		if !platforms.Only(p).Match(imgPlat) {
+			// Sometimes image variant is not populated due to legacy builders
+			// We still should support falling back here.
+			if imgPlat.OS == platform.OS && imgPlat.Architecture == platform.Architecture && imgPlat.Variant == "" {
+				logrus.WithField("image", refOrID).WithField("platform", platforms.Format(p)).Debug("Image platform cpu variant is not populated, but otherwise it matches what was requested")
+				return
+			}
```

Comment:
Just to note it somewhere - this is due to a quirk in containerd's platform matching where `platforms.Normalize` is invoked on both the argument to `platforms.Only` _and_ the things it's being compared against, so `linux/arm` gets normalized to `linux/arm/v7` in both cases (and thus fails if you specify `--platform linux/arm/v6` on a single-arch, no-manifest-list image that only has `os` and `architecture`).

Ideally, we'd fix `platforms.Only` in containerd to handle this case better, but it's complicated. :sweat_smile: